File: data/multilang_comparison/program_cpp_vector/src/func48.cpp
#include "program_vector_cpp.hpp" 
Array* func48(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array13;
   if (pCounter > 0) {
      array13 = vars->data[--pCounter];
      array13->refC++;
      DEBUG_COPY(array13->id);
   } else {
      array13 = new Array();
      array13->size = 370;
      array13->refC = 1;
      array13->id = 13;
      array13->data = new unsigned int[array13->size]();
      DEBUG_NEW(array13->id);
   }
   return array13;
}

File: data/multilang_comparison/program_go/src/func48.go
package main
import "math/rand"
var _ = rand.Intn
func Func48(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array13 *Array
   if pCounter > 0 {
       array13 = vars.Data[pCounter-1]
       pCounter--
       array13.RefC++
   } else {
       array13 = &Array{
           Size: 370,
           RefC: 1,
           Id: 13,
           Data: make([]uint32, 370),
       }
   }
   return array13;
}

